gulpfile.js/tasks/changelog.js

```diff
@@ -22,56 +22,41 @@ var runCommand = function (cmd) {
   })
 }
 
-var getCurrentCommit = function (commit) {
-  return new Promise(function (resolve, reject) {
-    if (commit) {
-      resolve(commit)
-    } else {
-      var cmd = 'git rev-parse HEAD'
-      runCommand(cmd).then(resolve)
-    }
-  })
-}
-
-var getChangedFiles = function (commit) {
-  if (!commit) {
-    throw new Error('No commit given')
+function getFileChanges(path) {
+  if (!path) {
+    throw new Error('No path given')
   }
 
-  var ref = process.env.GIT_PREVIOUS_SUCCESSFUL_COMMIT || 'master'
+  let cmd = `git diff master -- ${path}`
 
-  var cmd = 'git diff --name-only ' + ref + ' ' + commit
   return runCommand(cmd)
 }
 
-var checkForChangelog = function (files) {
-  if (files.length && !files.includes('CHANGELOG.md')) {
-    throw new Error('No CHANGELOG.md update')
-  }
-
-  return true
+function verifyChangelogChanges(lines) {
+  return new Promise((resolve, reject) => {
+    if (!lines) {
+      reject(new Error('No CHANGELOG.md update'))
+    } else {
+      resolve(lines);
+    }
+  })
 }
 
-gulp.task('changelog', function (done) {
-  var commit = process.env.TRAVIS
-    ? process.env.TRAVIS_COMMIT
-    : process.env.GIT_COMMIT
-
-  getCurrentCommit(commit)
-    .then(getChangedFiles)
-    .then(checkForChangelog)
-    .then(function () {
+gulp.task('changelog', (done) => {
+  getFileChanges('CHANGELOG.md')
+    .then(verifyChangelogChanges)
+    .then(() => {
+      gutil.log(gutil.colors.green('CHANGELOG.md verified'))
```

Comment:
We don't need to explicitly log the success. If gulp doesn't throw an error it's a success.